ftpvl/fetchers.py

```diff
@@ -41,6 +41,188 @@ def get_evaluation(self) -> Evaluation:
+            try:
+                next_page_token = "&pageToken=" + evals_json["nextPageToken"]
+            except KeyError:
+                break
+
+        return meta_urls
+
+    def _download(self) -> List[Dict]:
+        """
+        Fetches data from Google Cloud Storage, returning a list of decoded meta.json dicts
+        corresponding to the builds of the eval_num evaluation.
+
+        Returns
+        -------
+        List[Dict]
+            A list of dictionaries that represent the decoded meta.json files
+            of each test result.
+
+        Raises
+        ------
+        ValueError
+            Raised when fetcher wont find any successful builds from eval_num
+
+        """
+        # get build numbers from eval_num
+        meta_urls = self._get_builds(self.eval_num)
+
+        # fetch build info and download 'meta.json'
+        data = []
+        for meta_url in meta_urls:
+            # get build info
+            resp = requests.get(
+                    meta_url,
+                    headers={"Content-Type": "application/json"},
+            )
+            if resp.status_code != 200:
+                print(
+                    "Warning:",
+                    f"Unable to get build {meta_url} file. Status code: {resp.status_code}",
+                )
+                continue
+            try:
+                data.append(resp.json())
+            except json.decoder.JSONDecodeError:
+                print("Warning:", f"Unable to decode build {meta_url}")
+
+        if len(data) == 0:
+            raise ValueError(f"Unable to get any successful builds from eval_num {self.eval_num}.")
+
+        return data
+
+    def _preprocess(self, data: List[Dict]) -> pd.DataFrame:
+        """
+        Using data from _download(), processes and standardizes the data and
+        returns a Pandas DataFrame.
+        """
+        flattened_data = [Helpers.flatten(x) for x in data]
+
+        processed_data = []
+        for row in flattened_data:
+            legacy_icestorm = self._check_legacy_icebreaker(row)
+            processed_row = {}
+            if self.mapping is None:
+                processed_row = row
+            else:
+                for in_col_name, out_col_name in self.mapping.items():
+                    processed_row[out_col_name] = row[in_col_name]
+
+            actual_freq = Helpers.get_actual_freq(row, self.hydra_clock_names)
+            if actual_freq:
+                # freq in MHz, no change needed, newest fpga-tool-perf build reports freq in MHz
+                #TODO: change old builds from Hz to MHz
+                processed_row["freq"] = actual_freq
+            else:
+                processed_row["freq"] = 0.0
+            processed_row.update(Helpers.get_versions(row))
+            processed_data.append(processed_row)
+
+        return pd.DataFrame(processed_data).dropna(axis=1, how="all")
+
+    def _check_legacy_icebreaker(self, row):
```

Comment:
Given that now all frequencies are expressed in Hz, is this function still useful?

Reply:
No, it isn't. I removed it.